### PR TITLE
docs: Add Sphinx based docs

### PR DESCRIPTION
**Pull request by matthewfeickert**
Requires PR #248 to go in first.

Do not merge this as is, as there is some debug info in here that needs to get cleaned up.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+.. toctree::
+   :hidden:
+
+   intro
+   api
+   development
+
+.. include:: ../README.md
+   :parser: myst_parser.sphinx_
+   :end-before: <!-- Comment: Index page ends here in the docs -->
```
Comment (matthewfeickert) [inline docs/index.rst:22]: Generating and then _including_ rst files from Markdown is thanks to the amazing MyST project.

* https://www.sphinx-doc.org/en/master/usage/markdown.html
* https://myst-parser.readthedocs.io/en/latest/faq/index.html#include-markdown-files-into-an-rst-file

**Comment by ponyisi** [comment]
Hi @matthewfeickert - since we now do have a sphinx-based docs setup, is this PR still relevant?

**Comment by matthewfeickert** [comment]
> since we now do have a sphinx-based docs setup, is this PR still relevant?

@ponyisi  If there are docs then no, this PR should get closed out. But the docs are not linked anywhere on the [README](https://github.com/ssl-hep/ServiceX_frontend/blob/5bc9037e19016afc8adeaea211e4df01bb87071c/README.md) on either `master` or `3.0_develop` branch, the repository side bars, or the PyPI metadata

https://github.com/ssl-hep/ServiceX_frontend/blob/5bc9037e19016afc8adeaea211e4df01bb87071c/pyproject.toml#L58

. This means users don't know they exist and won't find them! :grimacing: 

While this PR can get closed, my suggestion is to have a PR that:

* Uses the `Closes #249` syntax
* Adds docs information to the README
* Updates the project URL metadata 

https://github.com/ssl-hep/ServiceX_frontend/blob/5bc9037e19016afc8adeaea211e4df01bb87071c/pyproject.toml#L57-L63

**Comment by kyungeonchoi** [comment]
Thanks @matthewfeickert - the new version generates sphinx docs

**Pull request closed without merging**